modules/build/src/main/scala/scala/build/input/ScalaCliInvokeData.scala

```diff
@@ -0,0 +1,37 @@
+package scala.build.input
+
+/** Stores information about how the program has been evoked
+  *
+  * @param progName
+  *   the actual Scala CLI program name which was run
+  * @param subCommand
+  *   the sub-command that was invoked by user
+  */
+
+case class ScalaCliInvokeData(progName: String, subCommandName: String, subCommand: SubCommand)
+
+enum SubCommand:
+  case Default extends SubCommand
+  case Shebang extends SubCommand
+  case Other   extends SubCommand
+
+object ScalaCliInvokeData {
+
+  // Copied from https://github.com/scalacenter/bloop/blob/main/shared/src/main/scala/bloop/io/Environment.scala
+  private lazy val shebangCapableShells = Seq(
+    "/bin/sh",
+    "/bin/ash",
+    "/bin/bash",
+    "/bin/dash",
+    "/bin/mksh",
+    "/bin/pdksh",
+    "/bin/posh",
+    "/bin/tcsh",
+    "/bin/zsh",
+    "/bin/fish"
+  )
+
+  def isShebangCapableShell = Option(System.getenv("SHELL")) match
+    case Some(currentShell) if shebangCapableShells.exists(sh => currentShell.contains(sh)) => true
+    case _                                                                                  => false
```

Comment:
Move this functions to some utils object, for example to `scala.cli.internal.ProcUtil`.